.github/scripts/filter_changed_files_go_test.sh

```diff
@@ -2,36 +2,35 @@
 # Copyright (c) HashiCorp, Inc.
 # SPDX-License-Identifier: BUSL-1.1
 
+set -euo pipefail
 
 # Get the list of changed files
-files_to_check=$(git diff --name-only origin/$GITHUB_BASE_REF)
+files_to_check=$(git diff --name-only "$(git merge-base origin/$BRANCH HEAD~)"...HEAD $*)
 
 # Define the directories to check
 skipped_directories=("docs/" "ui/" "website/" "grafana/")
 
 # Initialize a variable to track directories outside the skipped ones
-other_directories=""
-trigger_ci=true
+skip_ci=false
 
-# # Loop through the changed files and find directories/files outside the skipped ones
-# for file_to_check in $files_to_check; do
-# 	file_is_skipped=false
-# 	for dir in "${skipped_directories[@]}"; do
-# 		if [[ "$file_to_check" == "$dir"* ]] || [[ "$file_to_check" == *.md && "$dir" == *"/" ]]; then
-# 			file_is_skipped=true
-# 			break
-# 		fi
-# 	done
-# 	if [ "$file_is_skipped" = "false" ]; then
-# 		other_directories+="$(dirname "$file_to_check")\n"
-# 		trigger_ci=true
-# 		echo "Non doc file(s) changed - triggered ci: $trigger_ci"
-# 		echo -e $other_directories
-# 		echo "trigger-ci=$trigger_ci" >>"$GITHUB_OUTPUT"
-# 		exit 0 ## if file is outside of the skipped_directory exit script
-# 	fi
-# done
+# Loop through the changed files and find directories/files outside the skipped ones
+for file_to_check in $files_to_check; do
+	file_is_skipped=false
+	for dir in "${skipped_directories[@]}"; do
+		if [[ "$file_to_check" == "$dir"* ]] || [[ "$file_to_check" == *.md && "$dir" == *"/" ]]; then
+			file_is_skipped=true
+			break
+		fi
+	done
+	if [ "$file_is_skipped" = "false" ]; then
+		echo -e $file_to_check
+		echo "Non doc file(s) changed - skip-ci: $skip_ci"
+		echo "skip-ci=$skip_ci" >>"$GITHUB_OUTPUT"
+		exit 0 ## if file is outside of the skipped_directory exit script
+	fi
+done
 
-# echo "Only doc file(s) changed - triggered ci: $trigger_ci"
-echo "Doc file(s) change detection is currently disabled - triggering ci"
-echo "trigger-ci=$trigger_ci" >>"$GITHUB_OUTPUT"
+skip_ci=true
+echo -e "$files_to_check"
+echo "Only doc file(s) changed - skip-ci: $skip_ci"
+echo "skip-ci=$skip_ci" >>"$GITHUB_OUTPUT"
```

Comment:
**nitpick:** I am actually surprised that this works as I would have expected setting GITHUB_OUTPUT to be in a sub shell and not exported to githubs global variables.

I would personally set the environment variable setting at the workflow/job level like similar outputs.